client/eventsSDL/InputSourceKeyboard.cpp

```diff
@@ -30,13 +31,21 @@ InputSourceKeyboard::InputSourceKeyboard()
 
 void InputSourceKeyboard::handleEventKeyDown(const SDL_KeyboardEvent & key)
 {
-	if(key.repeat != 0)
-		return; // ignore periodic event resends
-
 	if (SDL_IsTextInputActive() == SDL_TRUE)
 	{
+		if(key.keysym.sym == SDLK_v && isKeyboardCtrlDown()) {
+		        std::string clipboardContent = SDL_GetClipboardText();
+		        boost::erase_all(clipboardContent, "\r");
+		        boost::erase_all(clipboardContent, "\n");
+		        GH.events().dispatchTextInput(clipboardContent);
+		        return;
+	 	}
+
```

Comment:
1. Use tabs, not spaces
2. Place bracket on new line
3. Check SDL docs: https://wiki.libsdl.org/SDL2/SDL_GetClipboardText
you need to call SDL_free() on pointer returned by `SDL_GetClipboardText`

```suggestion
		if(key.keysym.sym == SDLK_v && isKeyboardCtrlDown()) 
		{
			char * clipboardBuffer = SDL_GetClipboardText();
			std::string clipboardContent = clipboardBuffer;
			boost::erase_all(clipboardContent, "\r");
			boost::erase_all(clipboardContent, "\n");
			GH.events().dispatchTextInput(clipboardContent);
			SDL_free(clipboardBuffer);
			return;
	 	}

```